betka.py

```diff
@@ -0,0 +1,244 @@
+like 'make betka TARGET=fedora DOWNSTREAM_BRANCH=f32'
+                """
+                print(msg)
+                self.cwt_config = "default.yaml"
+            else:
+                # Examples config for F34 is f34.yaml
+                # For more configs see
+                # https://github.com/sclorg/container-workflow-tool/tree/master/container_workflow_tool/config
+                self.cwt_config = f"{self.downstream_branch}.yaml"
+        else:
+            if not self.downstream_branch:
+                msg = """
+DOWNSTREAM_BRANCH is not specified.
+Examples:
+rhel-8.3.0 for RHEL8
+rhscl-3.6-rhel7 for RHEL7
+For more details ask pkubat@redhat.com or phracek@redhat.com
+                """
+                print(msg)
+                return False
+            self.cwt_command = "rhcwt"
+        git_config = Path.home() / ".gitconfig"
+        if not git_config.exists():
+            msg = f"""
+File {git_config} is mandatory for using CWT tool.
+Create it by commands:
+git config --global user.mail <your mail>
+git config --global user.name <your name>
+            """
+            print(msg)
+            return False
+        return True
+
+    def delete_generated_dirs(self, ver: str):
+        results_dir: Path = self.cur_dir / f"results-{ver}"
+        if results_dir.exists():
+            print(f"results-{ver} dir was previously generated. Delete it to generate new sources.")
+            shutil.rmtree(results_dir)
+
+    def get_valid_images(self, ver) -> List[str]:
+        if self.os_env != "fedora":
+            self.convert_branch_to_cwt_tool()
+        cmd = f"docker run -it --rm {self.cwt_docker_image} bash -c '{self.cwt_command}" \
+              f" --config={self.cwt_config} utils listupstream'"
+        docker_output = run_cmd(cmd, return_output=True, shell=True)
+        valid_images = []
+        for line in docker_output.split('\n'):
+            # Fields are:
+            # python-27 python-27 s2i-python-container https://github.com/sclorg/s2i-python-container.git 2.7 rhel-8.4.0
+            if self.upstream_image_name in line and ver == line.split(" ")[4]:
+                valid_images.append(line)
+        print(f"Valid images {valid_images}")
+        return valid_images
+
+    def convert_branch_to_cwt_tool(self):
+        fields = self.downstream_branch.split('-')
+        if self.os_env == "rhel7":
+            self.cwt_config = f"rhel7.yaml:{fields[0]}{fields[1].replace('.','')}0"
+        else:
+            release_fields = fields[1].split('.')
+            self.cwt_config = f"rhel8.yaml:{fields[0]}{release_fields[0]}.{release_fields[1]}"
+
+    def clone_and_switch_to_branch(self, downstream_name: str, branch_name: str):
+        cmd = f"git clone --branch {branch_name} {self.clone_url}/{downstream_name} {self.betka_tmp_dir.name}/results"
+        run_cmd(cmd, shell=True)
+
+    def copy_upstream_sources(self):
+        p = Path(f"{self.betka_tmp_dir.name}/{self.upstream_image_name}")
```

Comment:
Might be more readable:
```suggestion
        p = Path(self.betka_tmp_dir.name) / self.upstream_image_name
```